File: ios/UGames/Network/UserInfoEndpointService.swift
import Foundation

struct UserInfoEndpointService {
    let networkService: NetworkService

    func profile() async throws -> ProfileResponseDTO {
        try await networkService.execute(UserInfoRequest(), attempts: 1)
    }
}

struct UserInfoRequest: Request {
    typealias DTO = ProfileResponseDTO

    var path: String { "/games/api/catalogue/v2/user_info" }
}

File: android/app/src/main/kotlin/games/yandex/wrap/network/UserInfoEndpointService.kt
package games.yandex.wrap.network

import games.yandex.wrap.network.dtos.ProfileResponseDto

class UserInfoEndpointService(
    private val networkService: NetworkService? = null,
) {
    suspend fun profile(): ProfileResponseDto =
        requireNetworkService().execute(UserInfoRequest())

    private fun requireNetworkService(): NetworkService =
        requireNotNull(networkService) { "NetworkService is required to execute user info requests." }
}

class UserInfoRequest : Request<ProfileResponseDto> {
    override val path = "/games/api/user/passport"
    override val serializer = ProfileResponseDto.serializer()
}
